transform/transformers/organisation.py

```diff
@@ -0,0 +1,161 @@
+# pylint: disable=line-too-long, wildcard-import, invalid-name, unused-wildcard-import
+"""Transform organisations"""
+import json
+from pyspark.sql import SparkSession, DataFrame
+from pyspark.sql.functions import lit
+from pyspark.sql.types import (
+    StringType,
+    ArrayType,
+    StructType,
+    StructField,
+)
+from transformers.base.base import BaseTransformer
+from schemas.properties.data import *
+from utils.utils import sort_schema
+
+
+class OrganisationTransformer(BaseTransformer):
+    """Transformer used to transform organisations"""
+
+    def __init__(self, spark: SparkSession):
+        self.type = "organisation"
+
+        super().__init__(
+            self.type, self.cols_to_add, self.cols_to_drop, self.cols_to_rename, spark
+        )
+
+    def apply_simple_trans(self, df: DataFrame) -> DataFrame:
+        """Apply simple transformations.
+        Simple in a way that there is a possibility to manipulate the main dataframe
+        without a need to create another dataframe and merging"""
+        df = df.withColumn(TYPE, lit(self.type))
+        df = self.rename_cols(df)
+
+        return df
+
+    def apply_complex_trans(self, df: DataFrame) -> DataFrame:
+        """Harvest oag properties that requires more complex transformations
+        Basically from those harvested properties there will be created another dataframe
+        which will be later on merged with the main dataframe"""
+        self.harvest_alternative_names(df, self.harvested_properties)
+        self.harvest_country(df, self.harvested_properties)
+        self.harvest_pids(df, self.harvested_properties)
+
+        return df
+
+    @staticmethod
+    def cast_columns(df: DataFrame) -> DataFrame:
+        """Cast columns"""
+        return df
+
+    @property
+    def harvested_schema(self) -> StructType:
+        """Schema of harvested properties"""
+        return sort_schema(
+            StructType(
+                [
+                    StructField(ALTERNATIVE_NAMES, ArrayType(StringType()), True),
+                    StructField(COUNTRY, StringType(), True),
+                    StructField(PIDS, StringType(), True),
+                ]
+            )
+        )
+
+    @property
+    def cols_to_add(self) -> None:
+        """Add those columns to the dataframe"""
+        return None
+
+    @property
+    def cols_to_drop(self) -> tuple[str, ...]:
+        """Drop those columns from the dataframe"""
+        return (
+            "alternativenames",
+            "country",
+            "pid",
+        )
+
+    @property
+    def cols_to_rename(self) -> dict[str, str]:
+        """Columns to rename. Keys are mapped to the values"""
+        return {
+            "legalname": LEGAL_NAME,
+            "legalshortname": LEGAL_SHORT_NAME,
+            "websiteurl": URL,
+        }
+
+    @staticmethod
+    def harvest_alternative_names(df: DataFrame, harvested_properties: dict) -> None:
```

Comment:
Not sure why this method is necessary at all. I assume that this method creates a union of `ALTERNATIVENAMES`, `LEGAL_NAME`, `LEGAL_SHORT_NAME`. 
- Raw `alternativenames` seems to include all names in the first place. Isnt that the case? 
- Simple rename is not the only required transformation?

Reply:
Raw alternative names do not include legal names or legal short names. Instead, they mainly contain names in the language of the country where the organization is located. I added these names to the field to improve user querying, but I can remove them if necessary.

Reply:
Discussed offline. Agreed to create such alternative_names that consists only from an unique values (no title or abb)

Reply:
Done